### PR TITLE
#54: Enable callable dictionary with __invoke method without configuring it

### PR DESCRIPTION
**Pull request by melicerte**

### CODE REVIEW

**Comment by Nek-** [comment]
Thanks, this looks nice. Could you add a test? Maybe something like this?

```php
// spec/Knp/DictionaryBundle/Dictionary/Factory/CallableFactorySpec.php
class CallableFactorySpec extends ObjectBehavior
{
    // ...existing code...

    public function it_create_a_dictionary_for_invokable_service($container)
    {
        $config = [
            'service' => 'service.id',
        ];

        $service = new CallableService();
        $container->get('service.id')->willReturn($service);

        $dictionary = $this->create('yolo', $config);

        $dictionary->getName()->shouldBe('yolo');
        $dictionary->getValues()->shouldBe([
            'foo1' => 'bar1',
            'foo2' => 'bar2',
            'foo3' => 'bar3',
        ]);
    }
}
class CallableService
{
    public function __invoke()
    {
        return [
            'foo1' => 'bar1',
            'foo2' => 'bar2',
            'foo3' => 'bar3',
        ];
    }
}
```

**Comment by melicerte** [comment]
You're absolutely right @Nek- 
Tested it is :)

File: spec/Knp/DictionaryBundle/Dictionary/Factory/CallableFactorySpec.php [hunk 1]
```diff
@@ -60,3 +81,15 @@ public function getYolo()
...
+{
+    public function __invoke()
+    {
+        return [
+            'foo1' => 'bar1',
+            'foo2' => 'bar2',
+            'foo3' => 'bar3',
+        ];
+    }
+}
```
Comment (Nek-) [inline spec/Knp/DictionaryBundle/Dictionary/Factory/CallableFactorySpec.php]: Last little change, can you add a new line at end of file? :)

Comment (melicerte) [reply spec/Knp/DictionaryBundle/Dictionary/Factory/CallableFactorySpec.php]: It's done ;)

**Pull request merged**